### DISCUSSION TITLE
Task 8

### DISCUSSION THREAD

**Discussion by JordanEmil** [Part-II-Computational-Physics/.github#45] [General]
I have gotten Task 9 to work for SparseHamiltonian but it seems to be giving me trouble with LinearHamiltonian. 

My Task 8 is as follows:
def LinearHamiltonian(N, h=1):
    H = spsl.LinearOperator((2*N, 2*N), matvec=lambda psi: ActWithH(psi, N, h))
    return H

Upon checking, e.g print(LinearHamiltonian(2)) I get the output
<4x4 _CustomLinearOperator with dtype=float64>

**Comment by JNHallen** [comment]
Hi! The first thing you should check is if whether you get the correct eigenvalues for your Linear Hamiltonian (by using spsl.eigsh as before). Your function for task 8 looks correct (assuming you mean ($2^N$, $2^N$) rather than (2N, 2N)), but it may be that your function ActWithH has a problem.

**Reply by JNHallen** [reply]
If you are getting some error messages you can paste them here and it will make it easier for us to help you.
